internal/functions/efgs/upload-batch.go

```diff
@@ -0,0 +1,247 @@
+package efgs
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	efgsutils "github.com/covid19cz/erouska-backend/internal/functions/efgs/utils"
+	"github.com/covid19cz/erouska-backend/internal/logging"
+	"github.com/covid19cz/erouska-backend/internal/secrets"
+	"github.com/golang/protobuf/proto"
+	"io/ioutil"
+	"net/http"
+	urlutils "net/url"
+	"os"
+	"strconv"
+	"time"
+)
+
+//UploadBatch Called in CRON every 2 hours. Gets keys from database and upload them to EFGS.
+func UploadBatch(w http.ResponseWriter, r *http.Request) {
```

Comment:
I believe this should not return HTTP 200 when an error has happened. Please make some helper function on side , write HTTP 500 with reasonable error message inside and call the function from all places where you use `return`.